scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/NovaseqXProcessor.java

```diff
@@ -0,0 +1,241 @@
+package ca.on.oicr.gsi.runscanner.scanner.processor;
+
+import ca.on.oicr.gsi.runscanner.dto.IlluminaDragenNotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.IlluminaNotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.NotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.type.AnalysisStatus;
+import ca.on.oicr.gsi.runscanner.dto.type.DRAGENWorkflow;
+import ca.on.oicr.gsi.runscanner.scanner.processor.NovaseqXProcessor.BCLConvertAnalysis.Analysis;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TimeZone;
+
+public class NovaseqXProcessor extends DefaultIllumina {
+  static class BCLConvertAnalysis {
+    List<Analysis> analyses = new LinkedList<>();
+
+    static class Analysis {
+      String sample;
+      int lane;
+      String read1File, read2File;
+      long readCount;
+
+      boolean isEmpty() {
+        return sample == null
+            && lane == 0
+            && read1File == null
+            && read2File == null
+            && readCount == 0;
+      }
+
+      ObjectNode toJson() {
+        ObjectNode analysisJson = MAPPER.createObjectNode();
+        analysisJson.put("Sample", sample);
+        analysisJson.put("Lane", lane);
+        analysisJson.put("Read1File", read1File);
+        analysisJson.put("Read2File", read2File);
+        analysisJson.put("ReadCount", readCount);
+        return analysisJson;
+      }
+    }
+
+    Analysis get(String sample, String lane) {
+      return get(sample, Integer.parseInt(lane));
+    }
+
+    Analysis get(String sample, int lane) {
+      return analyses
+          .stream()
+          .filter(a -> a.sample.equals(sample) && a.lane == lane)
+          .findFirst()
+          .orElse(new Analysis());
+    }
+
+    void put(Analysis newAnalysis) {
+      Analysis oldAnalysis = get(newAnalysis.sample, String.valueOf(newAnalysis.lane));
+      if (!oldAnalysis.isEmpty()) {
+        analyses.remove(oldAnalysis);
+      }
+      analyses.add(newAnalysis);
+    }
+
+    ArrayNode toJson() {
+      ArrayNode ret = MAPPER.createArrayNode();
+      analyses
+          .stream()
+          .sorted(
+              (a1, a2) -> {
+                int comp = a1.sample.compareTo(a2.sample);
+                if (comp == 0) { // same sample
+                  return Integer.compare(a1.lane, a2.lane);
+                }
+                return comp;
+              })
+          .forEach(a -> ret.add(a.toJson()));
+      return ret;
+    }
+  }
+
+  final String NUMERAL = "\\d+";
```

Comment:
Should be private. Several other fields and methods within the PR missing access modifiers where package private (default) doesn't make the most sense